### DISCUSSION TITLE
Update the process architecture image

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#2596] [General]
![image](https://user-images.githubusercontent.com/42310255/180593475-3aa221f8-772d-4a8f-818d-44970f0ea634.png)

主页这张图该更新了吧？现在find_package接口已经废弃，应该改成add_extsources/on_fetch/on_install。另外建议加一个xmake构建过程的时序图，on_check/on_config/on_load/on_build/on_link/on_install接口太多，容易分不清各个过程在哪个阶段，然后没有时序图也不容易理解xmake.lua为啥会被执行4次

**Comment by waruqi** [comment]
> xmake.lua为啥会被执行4次

不同阶段 解析 option/target/.. 域，还得判断 has_config 等状态，解析不同配置，每次解析依赖 lua 加载，这个目前只能这样，所以描述域不要做复杂脚本逻辑

**Reply by waruqi** [reply]
这块等回头有空吧，最近也没啥时间画图
